kafka-protocol-binding.md

```diff
@@ -95,28 +94,29 @@ value is made available as [UTF-8][RFC3629] encoded JSON text.
 
 ## 3. Kafka Message Mapping
 
-With Kafka 0.11.0.0 and above, the content mode is chosen by the sender of the 
+With Kafka 0.11.0.0 and above, the content mode is chosen by the sender of the
 event. Protocol usage patterns that might allow solicitation of events using a
 particular content mode might be defined by an application, but are not defined
 here.
 
 The receiver of the event can distinguish between the two content modes by
-inspecting the `ce_datacontenttype` [Header][Kafka-Message-Header] of the 
+inspecting the `content-type` [Header][Kafka-Message-Header] of the
 Kafka message. If the value is prefixed with the CloudEvents media type
 `application/cloudevents`, indicating the use of a known
-[event format](#14-event-formats), the receiver uses *structured* mode, otherwise
-it defaults to *binary* mode.
+[event format](#14-event-formats), the receiver uses *structured* mode,
```

Comment:
What's the intended behavior if there is *no* `content-type` header at all? We cannot default to binary in that case, as then things wouldn't work with AK < 0.11 at all (no header support as you described above). I tried to address that in my original PR by saying if the header property isn't provided, then the receiver uses structured mode with the JSON event format.

So how about adding this bit to make this clear:

> If no value is present, the receiver uses *structured* mode with the JSON event format. If the value is prefixed with the CloudEvents media type...

Reply:
What do people normally do when an Kafka message comes in w/o a content-type? Is it totally undefined and they're on their own? For AK < 0.11 we require things to use structured+JSON - would it be horrible to do the same when content-type is there for AK > 0.11 (meaning require JSON)?

Reply:
So actually this is what made stumble upon all this when reading it initally. AFAIK, it's not that `content-type` is a well-defined or commonly used header property in Kafka (of course I might be wrong and just not have heard about it). Really message key and value are binary data in Kafka and producers and consumers must agree on what format is used for a specific topic. So yes, if consumers don't have the knowledge about the content type either by means of such header property or out-of-bands exchange and agreement of the content type, they are on their own.

JSON seems like a reasonable default to me in the context of CloudEvents. The expectation is (AFAICS) that most people by now are on Kafka versions with header support, making the problem disappear.

Reply:
ok see what you think of my latest edits

Reply:
Yes, all looking good to me now. Thanks!

Maybe just on 3.3.1 though:

> The Kafka message header property content-type MUST be set to the media type of an event format.

It MUST be set to such value when it actually is given, but it's not that it MUST be given at all. I assume that's what you mean, but it currently may be understood as if that header property is mandatory, which it isn't.

Perhaps this: "If given, the Kafka message header property..."?

Also fine for me if you think that the current wording actually doesn't preclude it being not given.

Reply:
done!

Reply:
Cool, this PR LGTM then. Thanks!